pkg/api/server/v1alpha2/records_test.go

```diff
@@ -245,6 +246,258 @@ func TestGetRecord(t *testing.T) {
+				Name: fmt.Sprintf("%s/records/%d", result.GetName(), i),
+				Data: &pb.Any{
+					Type: "TaskRun",
+					Value: jsonutil.AnyBytes(t, &v1beta1.TaskRun{ObjectMeta: v1.ObjectMeta{
+						Name: fmt.Sprintf("%d", i),
+					}}),
+				},
+			},
+		})
+		if err != nil {
+			t.Fatalf("could not create result: %v", err)
+		}
+		t.Logf("Created record: %+v", r)
+		records = append(records, r)
+	}
+
+	// Create 3 PipelineRun records
+	for i := 3; i < 6; i++ {
+		fakeClock.Advance(time.Second)
+		r, err := srv.CreateRecord(ctx, &pb.CreateRecordRequest{
+			Parent: result.GetName(),
+			Record: &pb.Record{
+				Name: fmt.Sprintf("%s/records/%d", result.GetName(), i),
+				Data: &pb.Any{
+					Type: "PipelineRun",
+					Value: jsonutil.AnyBytes(t, &v1beta1.PipelineRun{ObjectMeta: v1.ObjectMeta{
+						Name: fmt.Sprintf("%d", i),
+					}}),
+				},
+			},
+		})
+		if err != nil {
+			t.Fatalf("could not create result: %v", err)
+		}
+		t.Logf("Created record: %+v", r)
+		records = append(records, r)
+	}
+
+	reversedRecords := make([]*pb.Record, len(records))
+	for i := len(reversedRecords); i > 0; i-- {
+		reversedRecords[len(records)-i] = records[i-1]
+	}
+
+	tt := []struct {
+		name   string
+		req    *pb.ListRecordsRequest
+		want   *pb.ListRecordsResponse
+		status codes.Code
+	}{
+		{
+			name: "all",
+			req: &pb.ListRecordsRequest{
+				Parent: result.GetName(),
+			},
+			want: &pb.ListRecordsResponse{
+				Records: records,
+			},
+		},
+		{
+			name: "list all records without knowing the result name",
+			req: &pb.ListRecordsRequest{
+				Parent: "foo/results/-",
+			},
+			want: &pb.ListRecordsResponse{
+				Records: records,
+			},
+		},
+		{
+			name: "list all records without knowing the parent and the result name",
+			req: &pb.ListRecordsRequest{
+				Parent: "-/results/-",
+			},
+			want: &pb.ListRecordsResponse{
+				Records: records,
+			},
+		},
+		{
+			// TODO: We should return NOT_FOUND in the future.
+			name: "missing parent",
+			req: &pb.ListRecordsRequest{
+				Parent: "foo/results/baz",
+			},
+			status: codes.NotFound,
+		},
+		{
+			name: "filter by record property",
+			req: &pb.ListRecordsRequest{
+				Parent: result.GetName(),
+				// Filter: `name == "foo/results/bar/records/0"`,
+				Filter: `name == "0"`,
+			},
+			want: &pb.ListRecordsResponse{
+				Records: records[:1],
+			},
+		},
+		{
+			name: "filter by record data",
+			req: &pb.ListRecordsRequest{
+				Parent: result.GetName(),
+				Filter: `data.metadata.name == "0"`,
+			},
+			want: &pb.ListRecordsResponse{
+				Records: records[:1],
+			},
+		},
+		{
+			name: "filter by record type",
+			req: &pb.ListRecordsRequest{
+				Parent: result.GetName(),
+				Filter: `data_type == "TaskRun"`,
+			},
+			want: &pb.ListRecordsResponse{
+				Records: records[:3],
+			},
+		},
+		{
+			name: "filter by parent",
+			req: &pb.ListRecordsRequest{
+				Parent: "-/results/-",
+				Filter: fmt.Sprintf(`parent + "/results/" + result_name == %q`, result.GetName()),
```

Comment:
The `parent` is also not the same. It used to be `foo/results/bar` instead of only `foo`. I tried to replicate that using string concatenation but then I realised this was not compiling correctly in cel2sql package.